kpi/serializers/v2/service_usage.py

```diff
@@ -172,30 +175,29 @@ def _get_current_year_start_date(self):
             return self._anchor_date.replace(year=self._now.year - 1)
         return self._anchor_date.replace(year=self._now.year)
 
+    def _filter_by_user(self, counter_query: QuerySet) -> QuerySet:
+        """
+        Filter counter models by a list of user ids
+        """
+        return counter_query.filter(user_id__in=self._user_ids)
+
     def _get_nlp_user_counters(self, month_filter, year_filter):
-        nlp_tracking = (
-            NLPUsageCounter.objects.only(
-                'date', 'total_asr_seconds', 'total_mt_characters'
-            )
-            .filter(
-                user_id__in=self._user_ids,
-            )
-            .aggregate(
-                asr_seconds_current_year=Coalesce(
-                    Sum('total_asr_seconds', filter=year_filter), 0
-                ),
-                mt_characters_current_year=Coalesce(
-                    Sum('total_mt_characters', filter=year_filter), 0
-                ),
-                asr_seconds_current_month=Coalesce(
-                    Sum('total_asr_seconds', filter=month_filter), 0
-                ),
-                mt_characters_current_month=Coalesce(
-                    Sum('total_mt_characters', filter=month_filter), 0
-                ),
-                asr_seconds_all_time=Coalesce(Sum('total_asr_seconds'), 0),
-                mt_characters_all_time=Coalesce(Sum('total_mt_characters'), 0),
-            )
+        nlp_tracking = NLPUsageCounter.objects.only('date', 'total_asr_seconds', 'total_mt_characters')
```

Comment:
What's going on here, aren't we just redefining the same variable?

Reply:
We're just chain-filtering the queryset. I've rewritten it to be a bit clearer.